internal/cmd/server.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/envoyproxy/gateway/api/config/v1alpha1"
+	"github.com/envoyproxy/gateway/pkg/envoygateway"
+	"github.com/envoyproxy/gateway/pkg/provider/kubernetes"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var configPath string
+
+// getServerCommand returns the server cobra command to be executed.
+func getServerCommand(log logr.Logger) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "server",
+		Aliases: []string{"serve"},
+		Short:   "Serve Envoy Gateway",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return serve(log)
+		},
+	}
+	cmd.PersistentFlags().StringVarP(&configPath, "config-path", "c", "/etc/envoy-gateway/config.yaml",
+		"The path to the configuration file.")
+
+	return cmd
+}
+
+// serve Envoy Gateway using log as the logger.
+func serve(log logr.Logger) error {
+	cfg := &envoygateway.Config{Logger: log}
+
+	// Load the config file.
+	data, err := os.ReadFile(configPath)
```

Comment:
with the current approach the process will exit early if a user only runs `envoy-gateway server` and `/etc/envoy-gateway/config.yaml` is missing . Instead should `config-path` default to "", and if empty serve can use system defaults ?